### PR TITLE
Switch from samuel/go-zookeeper to talbright/go-zookeeper, update ZK Lock.

### PR DESCRIPTION
**Pull request by kenbreeman**
@talbright thoughts?

### CODE REVIEW

File: Godeps/Godeps.json [hunk 1]
```diff
@@ -410,8 +410,8 @@
 			"Rev": "983d3a5fab1bf04d1b412465d2d9f8430e2e917e"
 		},
 		{
-			"ImportPath": "github.com/samuel/go-zookeeper/zk",
-			"Rev": "5250732bd2ed71d1e374212ebfc32760eca10c0a"
+			"ImportPath": "github.com/talbright/go-zookeeper/zk",
+			"Rev": "7cf7df1049b6283e64241060c9a92dea32437fc7"
```
Comment (talbright) [inline Godeps/Godeps.json:414]: Note that last night I merged in one more feature (talbright/go-zookeeper/pull/9), but its not a bug fix, just a small enhancement.

File: physical/zookeeper.go [hunk 2]
```diff
@@ -317,7 +317,7 @@ func (i *ZookeeperHALock) Lock(stopCh <-chan struct{}) (<-chan struct{}, error)
 func (i *ZookeeperHALock) attemptLock(lockpath string, didLock chan struct{}, failLock chan error, releaseCh chan bool) {
 	// Wait to acquire the lock in ZK
 	lock := zk.NewLock(i.in.client, lockpath, i.in.acl)
-	err := lock.Lock()
+	_, err := lock.Lock([]byte{})
```
Comment (talbright) [inline physical/zookeeper.go:320]: Yup, this is the only backwards incompatible change I can think of. I should probably add some window dressing like:

``` go
var ZkNoData = []byte{0}
```

**Comment by talbright** [comment]
I know that per the discussion in hashicorp/vault/issues/1367, your main issue was with a panic fix due to a buffer overflow (which has been merged in.) Is that a panic you can easily reproduce, to further verify the fix?

**Comment by kenbreeman** [comment]
Very easy to reproduce, just need 150,000+ tokens in use and then attempt to restart Vault :smile: 

I'm still working on testing this out so it may be a few days before I PR it upstream.

**Comment by talbright** [comment]
Unwitting beta testers 😁

Let me know if anything comes up.

**Comment by talbright** [comment]
You know, I woke up this morning and realized, I probably should have left the `Lock` function alone, and created a new one. At the time I made that change, I wasn't planning on maintaining go-zookeeper, so I wasn't worried about backwards compatibility. While `lock.go` could use a number of enhancements, I don't think this small change justifies breaking backwards compatibility.

The change I would make would be something like:

``` go
func (l *Lock) Lock() error {
  _,err:=l.LockWithData([]byte{0})
 return err
}
func (l *Lock) LockWithData(data []byte) (string, error) {
  //function body omitted...
}
```

Then you wouldn't have to monkey with updating lock calls.

If that sounds good to you, hold off on this PR, let me make that change and you can bump the version in Godeps.

**Comment by kenbreeman** [comment]
Sounds good, I'll pull in the 0.1.0 release you just made and test it out.

**Pull request closed without merging**